p2p/libp2p/netMessenger.go

```diff
@@ -236,6 +236,61 @@ func (netMes *networkMessenger) sendMessage() {
 	}
 }
 
+func (netMes *networkMessenger) createConnectionMonitor(targetConnCount int) error {
+	var err error
+
+	reconnecter, _ := netMes.peerDiscoverer.(p2p.Reconnecter)
+	netMes.connMonitor, err = newConnectionMonitor(
+		reconnecter,
+		netMes.ctxProvider,
+		defaultThresholdMinConnectedPeers,
+		targetConnCount,
+	)
+	if err != nil {
+		return err
+
+	}
+	notifee := &connectionMonitorNotifiee{
+		ConnectionMonitor: netMes.connMonitor,
+	}
+	netMes.ctxProvider.connHost.Network().Notify(notifee)
+
+	go func() {
+		for {
+			netMes.connMonitor.DoReconnectionBlocking()
+			time.Sleep(durationBetweenReconnectionAttempts)
+		}
+	}()
+
+	go func() {
+		for {
+			netMes.connMonitor.CheckConnectionsBlocking()
+			time.Sleep(durationCheckConnections)
+		}
+	}()
+
+	return nil
+}
+
+// ApplyOptions can set up different configurable options of a networkMessenger instance
+func (netMes *networkMessenger) ApplyOptions(opts ...p2p.Option) error {
+	cfg := &p2p.Config{}
+
+	for _, opt := range opts {
+		err := opt(cfg)
+		if err != nil {
+			return err
+		}
+	}
+
+	err := netMes.ctxProvider.SetPeerBlacklist(cfg.BlacklistHandler)
```

Comment:
this is strange. Can't this be done at constructor ? options are setting other stuff.

Reply:
The messenger can run without a working blacklist handler. Since the messenger's constructor parameter list keeps growing and growing I think it is better to change to WithOption pattern.